src/main/java/com/google/sps/servlets/Caption.java

```diff
@@ -38,8 +38,13 @@ private String parseXmlFromStream(InputStream inputStream)
     NodeList textDom = doc.getElementsByTagName("text");
     StringBuilder caption = new StringBuilder();
     for (int i = 0; i < textDom.getLength(); i++) {
-      caption.append(textDom.item(i).getFirstChild().getNodeValue());
-      caption.append(" ");
+      try {
+        caption.append(textDom.item(i).getFirstChild().getNodeValue());
```

Comment:
Why can it be null in the first place?

Reply:
Not entirely sure how captions work, it seems like each line of caption has max duration and it is an empty item if no one speaks during some period of time.